sw/device/lib/crypto/drivers/kmac.c

```diff
@@ -13,20 +13,242 @@
 
 // For each x (denoting security strength) in {128, 224, 256, 384, 512}
 // keccak_rates = (1600 - 2x) in given order
-static uint32_t keccak_rates[] = {168, 144, 136, 104, 72};
+static const size_t keccak_rates_lookup[] = {168, 144, 136, 104, 72};
+OT_ASSERT_ENUM_VALUE(kKmacSecurityStrength128, 0);
+OT_ASSERT_ENUM_VALUE(kKmacSecurityStrength224, 1);
+OT_ASSERT_ENUM_VALUE(kKmacSecurityStrength256, 2);
+OT_ASSERT_ENUM_VALUE(kKmacSecurityStrength384, 3);
+OT_ASSERT_ENUM_VALUE(kKmacSecurityStrength512, 4);
+
+// For each x in kmac_key_len_t, return the actual byte length of the key
+static const size_t key_len_lookup[] = {16, 24, 32, 48, 64};
+OT_ASSERT_ENUM_VALUE(kKmacKeyLength128, 0);
+OT_ASSERT_ENUM_VALUE(kKmacKeyLength192, 1);
+OT_ASSERT_ENUM_VALUE(kKmacKeyLength256, 2);
+OT_ASSERT_ENUM_VALUE(kKmacKeyLength384, 3);
+OT_ASSERT_ENUM_VALUE(kKmacKeyLength512, 4);
+
+enum {
+  kKmacPrefixRegCount = 4 * KMAC_PREFIX_MULTIREG_COUNT,
+  // The total size of prefix registers (in bytes), after removing len encodings
+  kKmacPrefixMaxSize = 4 * KMAC_PREFIX_MULTIREG_COUNT - 8,
+  kKmacBaseAddr = TOP_EARLGREY_KMAC_BASE_ADDR,
+  kKmacCfgAddr = kKmacBaseAddr + KMAC_CFG_SHADOWED_REG_OFFSET,
+  kKmacKeyShare0Addr = kKmacBaseAddr + KMAC_KEY_SHARE0_0_REG_OFFSET,
+  kKmacKeyShare1Addr = kKmacBaseAddr + KMAC_KEY_SHARE1_0_REG_OFFSET,
+
+  kKmacPrescalerMax = 0x3ff,
+  kKmacWaitTimerMax = 0xffff,
+  kKmacEntropyThresholdMax = 0x3ff,
+};
+
+OT_ASSERT_ENUM_VALUE(kKmacPrescalerMax, KMAC_ENTROPY_PERIOD_PRESCALER_MASK);
+OT_ASSERT_ENUM_VALUE(kKmacWaitTimerMax, KMAC_ENTROPY_PERIOD_WAIT_TIMER_MASK);
+OT_ASSERT_ENUM_VALUE(kKmacEntropyThresholdMax,
+                     KMAC_ENTROPY_REFRESH_THRESHOLD_SHADOWED_THRESHOLD_MASK);
```

Comment:
Because register headers do not define max value for these fields, I am using the `_MASK` values here (kind of misusing). `_MASK` values should match max values for these fields as long as they are 2^k-1. In short, I was thinking maybe it is a good idea to put these additional enum comparisons, so that if these mask values are modified later,  it automatically fails and we need to check if this assumption still holds.

Reply:
Come to think of it, the only way `_MASK` doesn't match max values are those weird cases where the bit fields are not continuous chunks of a register. Violating that continuity would also break `bitfield_field32_write`, so I guess that is enough guarantee for me to ignore those cases. Just use `_MASK` as max without hesitation is OK I think. In short, I will apply the changes here.

Reply:
Eventually I removed all three `...Max` values and put inline `UINT32_MAX` instead.

Reply:
That sounds like good reasoning; I agree there's not a clean way to get a max value here in all cases, but assuming bitfields will be continuous seems fair.